src/System.Management.Automation/engine/LanguagePrimitives.cs

```diff
@@ -2886,8 +2910,16 @@ private static bool TryScanNumber(string strToConvert, Type resultType, out obje
         {
             try
             {
+                var parsedNumber = Parser.ScanNumber(strToConvert, resultType, shouldTryCoercion: false);
+                if (resultType == typeof(BigInteger) || parsedNumber is BigInteger)
```

Comment:
What if we request `Int` (resultType) but get `BigInteger` in `parsedNumber`? Should we return `false`?

Reply:
No. `[int]"20s"` parses correctly (we get `short` for `parsedNumber` but the `resultType` desired there is `int`), so too should things like `[long]"10n"` (`BigInteger` for `parsedNumber` but `resultType` of `long`)